staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -106,10 +107,16 @@ func PatchResource(r rest.Patcher, scope RequestScope, admit admission.Interface
 		)
 
 		userInfo, _ := request.UserFrom(ctx)
-		staticAdmissionAttributes := admission.NewAttributesRecord(nil, nil, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Update, userInfo)
 		admissionCheck := func(updatedObject runtime.Object, currentObject runtime.Object) error {
-			if mutatingAdmission, ok := admit.(admission.MutationInterface); ok && admit.Handles(admission.Update) {
-				return mutatingAdmission.Admit(admission.NewAttributesRecord(updatedObject, currentObject, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Update, userInfo))
+			if mutatingAdmission, ok := admit.(admission.MutationInterface); ok {
+				if mutatingAdmission.Handles(admission.Create) {
+					if accessor, err := meta.Accessor(currentObject); err == nil && accessor.GetResourceVersion() == "" {
+						return mutatingAdmission.Admit(admission.NewAttributesRecord(updatedObject, nil, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Create, userInfo))
+					}
+				}
+				if mutatingAdmission.Handles(admission.Update) {
```

Comment:
Calling the admission for Update if Create is not created is odd. I could imagine that we want that for backwards compatibility (do we?), but then this must be well documented.

Reply:
@sttts Thanks! 

> I could imagine that we want that for backwards compatibility (do we?)

IMHO, it's not really for backwards compatibility, considering that the actual `update` is far more than `create-on-update`.  What about separating them into 2 anonymous `admissionCheck` functions? One for creation check and the other for original update check?

Reply:
@sttts A polite bump in this thread.😃

Reply:
@sttts Addressing reviews from @jennybuckley, all changes in `Patch` is reverted.

> We shouldn't change the patcher, because it doesn't currently allow create on update anyway.

Reply:
Where have we hardcoded that a patch cannot create an object? Have we?

Can we add a check which intentionally breaks if we or any 3rdparty apiserver creator somehow manage to create object via patch? Some like `// this code should never be reached. But if it is, we fail hard to be reminded that admission does not support this`.